koku/common/sql_retries.py

```diff
@@ -0,0 +1,72 @@
+import functools
+import logging
+import random
+import time
+import typing as t
+
+LOG = logging.getLogger(__name__)
+
+
+class TrinoQueryError(Exception):
+    pass
+
+
+def log_json(msg, log_ref, context):
+    return f"{msg} - {log_ref} - {context}"
+
+
+def retry(
+    original_callable: t.Callable = None,
+    *,
+    retries: int = 3,
+    retry_on: tuple[Exception, ...],
+    attribute_to_check: str = "message",
+    max_wait: int = 30,
+    context_extractor: t.Optional[t.Callable] = None,
```

Comment:
For now, I would do this directly in the function rather than making it a parameter.
```suggestion
```

Reply:
I may be wrong, but we cannot do that because we hit circular imports.

Reply:
You are correct. We need to rearrange some things to prevent a circular import.